test/Constraints/valid_pointer_conversions.swift

```diff
@@ -41,3 +41,9 @@ func SR12382(_ x: UnsafeMutablePointer<Double>??) {}
 var i = 0
 SR12382(&i) // expected-error {{cannot convert value of type 'UnsafeMutablePointer<Int>' to expected argument type 'UnsafeMutablePointer<Double>'}}
 // expected-note@-1 {{arguments to generic parameter 'Pointee' ('Int' and 'Double') are expected to be equal}}
+
+//problem/68254165 - Bad diagnostic when using String init(decodingCString:) with an incorrect pointer type
+func rdar68254165(ptr: UnsafeMutablePointer<Int8>) {
+  _ = String(decodingCString: ptr, as: .utf8) // expected-error {{generic parameter 'Encoding' could not be inferred}}
+  // expected-error@-1 {{type '_.Type' has no member 'utf8'}}
```

Comment:
New `HoleType` allows us to deal with this issue and avoid producing member diagnostic here, I'm going to implement that in a follow-up PR to separate concerns here...